libcst/metadata/scope_provider.py

```diff
@@ -86,6 +99,84 @@ class BuiltinAssignment(BaseAssignment):
     pass
 
 
+class QualifiedNameSource(Enum):
+    IMPORT = auto()
+    LOCAL = auto()
+
+
+@dataclass(frozen=True)
+class QualifiedName:
+    name: str
+    source: QualifiedNameSource
+
+
+class _QualifiedNameUtil:
+    @staticmethod
+    def get_full_name_for(node: cst.CSTNode) -> Optional[str]:
+        if isinstance(node, cst.Name):
+            return node.value
+        elif isinstance(node, cst.Attribute):
+            return (
+                f"{_QualifiedNameUtil.get_full_name_for(node.value)}.{node.attr.value}"
+            )
+        elif isinstance(node, cst.Call):
+            return _QualifiedNameUtil.get_full_name_for(node.func)
+        elif isinstance(node, cst.Subscript):
+            return _QualifiedNameUtil.get_full_name_for(node.value)
+        elif isinstance(node, cst.SimpleString):
+            # In the case of SimpleString of type hints.
+            return literal_eval(node.value)
+        return None
+
+    @staticmethod
+    def find_qualified_name_for_import_alike(
+        assignment_node: Union[cst.Import, cst.ImportFrom],
+        name_parts: List[str],
+        results: Set[QualifiedName],
```

Comment:
Its a bit weird that this takes in a set that is mutated instead of returning a new set.

Reply:
Fixed.